src/Tarcog/tst/units/GapLayerAtEdgeForcedVentilationWithZeroAirSpeed.unit.cpp

```diff
@@ -0,0 +1,182 @@
+#include <memory>
+#include <gtest/gtest.h>
+
+#include "WCETarcog.hpp"
+
+class TestGapLayerAtEdgeForcedVentilationWithZeroAirSpeed : public testing::Test
+{
+private:
+    std::shared_ptr<Tarcog::ISO15099::CSingleSystem> m_TarcogSystem;
+
+protected:
+    void SetUp() override
+    {
+        /////////////////////////////////////////////////////////
+        // Outdoor
+        /////////////////////////////////////////////////////////
+        auto airTemperature = 255.15;   // Kelvins
+        auto airSpeed = 5.5;            // meters per second
+        auto tSky = 255.15;             // Kelvins
+        auto solarRadiation = 0.0;
+
+        auto Outdoor = Tarcog::ISO15099::Environments::outdoor(
+          airTemperature, airSpeed, solarRadiation, tSky, Tarcog::ISO15099::SkyModel::AllSpecified);
+        ASSERT_TRUE(Outdoor != nullptr);
+        Outdoor->setHCoeffModel(Tarcog::ISO15099::BoundaryConditionsCoeffModel::CalculateH);
+
+        /////////////////////////////////////////////////////////
+        /// Indoor
+        /////////////////////////////////////////////////////////
+
+        auto roomTemperature = 295.15;
+
+        auto Indoor = Tarcog::ISO15099::Environments::indoor(roomTemperature);
+        ASSERT_TRUE(Indoor != nullptr);
+
+        // IGU
+        auto solidLayerThickness = 0.005715;   // [m]
+        auto solidLayerConductance = 1.0;
+
+        auto solidLayer = Tarcog::ISO15099::Layers::solid(solidLayerThickness, solidLayerConductance);
+        ASSERT_TRUE(solidLayer != nullptr);
+
+        auto shadeLayerThickness = 0.01;
+        auto shadeLayerConductance = 160.0;
+        auto Atop = 0.1;
+        auto Abot = 0.1;
+        auto Aleft = 0.1;
+        auto Aright = 0.1;
+        auto Afront = 0.2;
+
+        EffectiveLayers::ShadeOpenness openness{Afront, Aleft, Aright, Atop, Abot};
+
+        double windowWidth = 1;
+        double windowHeight = 1;
+
+        EffectiveLayers::EffectiveLayerOther effectiveLayer{
+          windowWidth, windowHeight, shadeLayerThickness, openness};
+
+        EffectiveLayers::EffectiveOpenness effOpenness{effectiveLayer.getEffectiveOpenness()};
+
+        auto shadeLayer = Tarcog::ISO15099::Layers::shading(
+          shadeLayerThickness, shadeLayerConductance, effOpenness);
+
+        ASSERT_TRUE(shadeLayer != nullptr);
+
+        auto gapThickness = 0.0127;
+        auto gapAirSpeed = 0.0;
```

Comment:
@bbuenoun In this test I set the gap air speed to `0`. Compared to [the test with the value `0.5`](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/files#diff-f1510538eafc05022ed4282de85aba1f6ce25356865aa7c328620229546650a7R67) only the gain energy changes from `-173.19394352133912` to `0.0`. The reason is that the way the computation currently works, we **force** an air speed of `0` no matter what in [`CIGUShadeLayer::calcEdgeShadeFlow`](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/files#diff-373f0b5a3bab54faaf37cfa2a370957a33143cf5af21a40aa39861baea9356e1R297-R300) (any possible natural ventilation is suppressed). And this air speed of `0` is [used as multiplicative factor in the calculation of `CIGUVentilatedGapLayer::ventilatedFlow`](https://github.com/ise-bipv/Windows-CalcEngine/blob/add-gaps-with-forced-ventilation/src/Tarcog/src/IGUVentilatedGapLayer.cpp#L182) resulting in the ventilated flow `0`.

Reply:
We have to compare the test with air_velocity=0 with a test without natural ventilation, i.e. sealed and narrow gap between two layers.

Reply:
In https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/4bda8f0a9a9848b4230f686345c518d9cd07ebdb I added a test with a naturally ventilated [narrow](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/4bda8f0a9a9848b4230f686345c518d9cd07ebdb#diff-77b34fc1601b2388bada4d0314de1afe488763836024690d16cc0a15b08d9f92R66) gap at the edge. The [gain flow](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/4bda8f0a9a9848b4230f686345c518d9cd07ebdb#diff-77b34fc1601b2388bada4d0314de1afe488763836024690d16cc0a15b08d9f92R116) is close to zero.

Reply:
I made a naive attempt to implement sealed forced-ventilation gaps in https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/156b5e035530a00945bf5bb65829df06d1fd703b I currently use the previous layer's temperature as inlet temperature for the gap. Should one be able to configure which layer is used for the inlet temperature? Or should it always be the previous or next layer? I added a test with a [normal sealed forced-ventilation gap](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/156b5e035530a00945bf5bb65829df06d1fd703b#diff-2bc71bc9c8ff5c69e24cd6a17dcce54e9424c2df4496d2976ae854b5a6354b4a), one [such that is narrow](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/156b5e035530a00945bf5bb65829df06d1fd703b#diff-f682c31f98199791424b8439fff92ca8201334d507c0f6f96b8d839cc7c143f2), and one [such that is narrow and has air speed zero](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/aca8734711b4c9721d6a919f5a92ea06fbe7dfd0). Which test should produce similar numbers as [GapLayerAtEdgeForcedVentilationWithZeroAirSpeed](https://github.com/ise-bipv/Windows-CalcEngine/blob/add-gaps-with-forced-ventilation/src/Tarcog/tst/units/GapLayerAtEdgeForcedVentilationWithZeroAirSpeed.unit.cpp)? [GapLayerSealedForcedVentilationNarrow](https://github.com/ise-bipv/Windows-CalcEngine/blob/add-gaps-with-forced-ventilation/src/Tarcog/tst/units/GapLayerSealedForcedVentilationNarrow.unit.cpp)?

Reply:
[GapLayerAtEdgeForcedVentilationWithZeroAirSpeed](https://github.com/ise-bipv/Windows-CalcEngine/blob/add-gaps-with-forced-ventilation/src/Tarcog/tst/units/GapLayerAtEdgeForcedVentilationWithZeroAirSpeed.unit.cpp) has to be compared with a configuration where all the gaps (Atop, Abottom, Afront, Aright, Aleft) are set to zero.

Reply:
Please compare [GapLayerAtEdgeForcedVentilationWithZeroAirSpeed](https://github.com/ise-bipv/Windows-CalcEngine/blob/add-gaps-with-forced-ventilation/src/Tarcog/tst/units/GapLayerAtEdgeForcedVentilationWithZeroAirSpeed.unit.cpp) with a configuration with the same geometry, naturally ventilated, where all the gaps (Atop, Abottom, Afront, Aright, Aleft) are set to zero.

Reply:
I [added a test for the requested configuration](https://github.com/ise-bipv/Windows-CalcEngine/pull/2/commits/5c6fdf145db1467262532c1768c4da5f1ddb8faa) (sorry for being so stubborn) and compared the numbers. The temperatures only differ in the third decimal place and the gain energy differs by `0.0182...`.